storage/url/url.go

```diff
@@ -278,9 +278,37 @@ func (u *URL) Clone() *URL {
 }
 
 // SetRelative explicitly sets the relative path of u against given base value.
-func (u *URL) SetRelative(base string) {
-	dir := filepath.Dir(base)
-	u.relativePath, _ = filepath.Rel(dir, u.Absolute())
+// if the base path contains `globCharacters` then, the
+// relative path is determined wrt the parent directory
+// of the so called wildcarded object.
```

Comment:
```suggestion
// If the base path contains `globCharacters` then, the relative path is
// determined with respect to the parent directory of the so called wildcarded
// object.
```
**Nitpick:**
Comments are usually 80 chars per line there is more space that we can use. Also, personally not a huge fan of ambiguous abbrevations like *wrt*, is is respect, regard or reference? 🙃